lightning/src/ln/channelmanager.rs

```diff
@@ -291,6 +291,32 @@ struct ReceiveError {
 	msg: &'static str,
 }
 
+/// This enum is used to specify which error data to send to peers when failing back an HTLC
+/// using [`ChannelManager::fail_htlc_backwards_with_reason`].
+///
+/// For more info on failure codes, see <https://github.com/lightning/bolts/blob/master/04-onion-routing.md#failure-messages>.
+#[derive(Clone, Copy)]
+pub enum FailureCode {
+	/// A general temporary failure for otherwise unspecified errors that occur
+	/// for the entire node while handling the received onion. This can be used in situations
+	/// where other temporary error codes would not fit, while indicating to the origin node
+	/// it may want to retry the payment.
+	TemporaryNodeFailure             = 0x2000 | 2,
+	/// We have a required feature which was not in this onion.
+	/// You may wish to use this in cases where you require some additional metadata be provided
+	/// with the payment but which was not provided here.
+	RequiredNodeFeatureMissing       = 0x4000 | 0x2000 | 3,
+	/// The CLTV expiry is too close to the current block height for safe handling by us. This
+	/// should only be used when failing back as a forwarding node.
+	ExpiryTooSoon                    = 0x1000 | 14,
+	/// You may wish to use this when a `payment_hash` is unknown, the `payment_secret` is
+	/// required and not present, the `payment_secret` doesn't match the `payment_hash`, the
+	/// received amount is incorrect, the CLTV expiry of the HTLC is too close to the current
+	/// block height for safe handling, or `payment_metadata` isn't present when it should be.
+	/// This failure code should be used when acting as the final node in a route.
```

Comment:
```suggestion
```
This error enum will only be used by final nodes :) Could also note that using this variant is equivalent to calling `fail_htlc_backwards`.